tlwatch/auction_website.py

```diff
@@ -0,0 +1,73 @@
+import functools
+import logging
+import re
+from hashlib import md5
+from urllib import parse
+
+import bernhard
+import click
+import requests
+from bs4 import BeautifulSoup
+
+from tlwatch import util
+
+logger = logging.getLogger(__name__)
+
+
+def calculate_website_source_hash(url: str) -> str:
+    """Fetch sources of the website and calculate hash.
+
+    Relevant sources are the HTML page and the JavaScript files. For the
+    latter ones, only the first party scripts are of interest, which are
+    stored at the `/js` directory. The hash is calculated over the
+    concatenation of all relevant sources. Be aware that this depends on
+    the order of the script imports. This is considered to be fine,
+    since reorder them would also change the HTML source.
+    """
+
+    source_list = []
+    html = requests.get(url).text
+    source_list.append(html)
+    soup = BeautifulSoup(html, features="html.parser")
+    script_url_list = [
+        parse.urljoin(url, script.get("src"))
+        for script in soup.findAll("script", attrs={"src": re.compile("^/js")})
+    ]
+
+    for script_url in script_url_list:
+        script = requests.get(script_url).text
+        source_list.append(script)
+
+    source_hash = md5("".join(source_list).encode())
+    return source_hash.hexdigest()
+
+
+def watch_auction_website(url: str, original_hash: str):
+    current_hash = calculate_website_source_hash(url)
```

Comment:
Please catch errors from requests here and report them as error to riemann. You can report with status "io-error" and report more details as "description" field.

Reply:
I've took you're error handling here as example. :heavy_check_mark: